src/hooks/useDeliverablePlanning.tsx

```diff
@@ -0,0 +1,55 @@
+import { getValueByYPath } from '@/lib/yUtils'
+import { useCallback, useEffect, useState } from 'react'
+import type * as Y from 'yjs'
+import { useCollaborationDocument } from './useCollaborationDocument'
+import type { EleBlock } from '@/shared/types'
+import { useDeliverablePlanningId } from './index/usePlanningIdFromAssignmentId'
+
+interface DeliverableReferences {
+  planningUuid: string
+  yRoot: Y.Map<unknown>
+  getAssignment: (deliverableType?: string) => { id: string | undefined, index: number }
+}
+
+/**
+ * This hook can receive a deliverableId (i.e. document uuid from an article/flash) and return
+ * the actual planning y document with information on which assignment in that planning document
+ * is referencing the deliverable.
+ */
+export const useDeliverablePlanning = (deliverableId: string): DeliverableReferences | null => {
+  const planningUuid = useDeliverablePlanningId(deliverableId)
+  const planningDoc = useCollaborationDocument({ documentId: planningUuid })
+  const [yRoot, setYRoot] = useState<Y.Map<unknown> | undefined>()
+
+  useEffect(() => {
+    if (planningDoc?.document) {
+      const ele = planningDoc.document.getMap('ele')
+      setYRoot((prev) => (prev !== ele) ? ele : prev)
```

Comment:
A bit unsure how this check works? ele is always recreated? and since its a reference type the value is always reset to ele.

Wouldn't this cause less renders:
```suggestion
      if (ele !== yRoot) {
        setYRoot(ele);
      }

```

Reply:
The `setYRoot((prev) => (prev !== ele) ? ele : prev)` is the correct way to do what you suggest. Your suggestion would force me to add yRoot and that could potentially cause infinite rerenders. You don't want to set a state variable in a hook that would rerun if that state variable changes.